### PR TITLE
1808: use a separate Properties instance for kafka consumer properties

### PR DESCRIPTION
**Pull request by sergeylemekhov**

### CODE REVIEW

**Comment by artembilan** [comment]
Thank you for contribution, @sergeylemekhov !

Please, take a look what @garyrussell has just PR'ed this morning #1809.

Looks like you both did similar fix.

Please, review and let me know which PR I should close as duplicated.

File: spring-kafka/src/main/java/org/springframework/kafka/config/AbstractKafkaListenerContainerFactory.java [hunk 1]
```diff
@@ -402,7 +403,8 @@ private void configureEndpoint(AbstractKafkaListenerEndpoint<K, V> aklEndpoint)
 	protected void initializeContainer(C instance, KafkaListenerEndpoint endpoint) {
 		ContainerProperties properties = instance.getContainerProperties();
 		BeanUtils.copyProperties(this.containerProperties, properties, "topics", "topicPartitions", "topicPattern",
-				"messageListener", "ackCount", "ackTime", "subBatchPerPartition");
+				"messageListener", "ackCount", "ackTime", "subBatchPerPartition", "kafkaConsumerProperties");
+		properties.setKafkaConsumerProperties(new Properties(this.containerProperties.getKafkaConsumerProperties()));
```
Comment (artembilan) [inline spring-kafka/src/main/java/org/springframework/kafka/config/AbstractKafkaListenerContainerFactory.java:407]: I comment out this line of code and no one test fails, even your new one.
What am I missing in the logic?
Thanks

**Review by artembilan — CHANGES_REQUESTED** [review]

File: spring-kafka/src/main/java/org/springframework/kafka/config/AbstractKafkaListenerContainerFactory.java (hunk 1 above)
Comment (garyrussell) [reply spring-kafka/src/main/java/org/springframework/kafka/config/AbstractKafkaListenerContainerFactory.java:407]: This code is not necessary; the constructor initializes the field with a new `Properties`.

Comment (artembilan) [reply spring-kafka/src/main/java/org/springframework/kafka/config/AbstractKafkaListenerContainerFactory.java:407]: OK. Merging your PR then, Gary.

**Comment by artembilan** [comment]
Duplicates #1809.

@sergeylemekhov , 

thank you for effort any way; looking forward for more contribution!

**Pull request closed without merging**